test/e2e_node/cpumanager_test.go

```diff
@@ -0,0 +1,1596 @@
+		mismatchedCPUs := expectedCPUs.Difference(cpus)
+		md.MismatchedCPUs = mismatchedCPUs.String()
+		return mismatchedCPUs.Size() <= toleration, nil
+	}).WithTemplate("Pod {{.Actual.Namespace}}/{{.Actual.Name}} UID {{.Actual.UID}} has allowed CPUs <{{.Data.CurrentCPUs}}> not all thread sibling pairs (would be <{{.Data.ExpectedCPUs}}> mismatched <{{.Data.MismatchedCPUs}}> toleration <{{.Data.Count}}>) for container {{.Data.Name}}", md)
+}
+
+// Other helpers
+
+func getContainerAllowedCPUs(pod *v1.Pod, ctnName string, isInit bool) (cpuset.CPUSet, error) {
+	cgPath, err := makeCgroupPathForContainer(pod, ctnName, isInit)
+	if err != nil {
+		return cpuset.CPUSet{}, err
+	}
+	framework.Logf("pod %s/%s cnt %s qos=%s path %q", pod.Namespace, pod.Name, ctnName, pod.Status.QOSClass, cgPath)
+	data, err := os.ReadFile(filepath.Join(cgPath, "cpuset.cpus"))
+	if err != nil {
+		return cpuset.CPUSet{}, err
+	}
+	cpus := strings.TrimSpace(string(data))
+	framework.Logf("pod %s/%s cnt %s cpuset %q", pod.Namespace, pod.Name, ctnName, cpus)
+	return cpuset.Parse(cpus)
+}
+
+func getSandboxCFSQuota(pod *v1.Pod) (string, error) {
+	cgPath := filepath.Join(makeCgroupPathForPod(pod), "cpu.max")
+	data, err := os.ReadFile(cgPath)
+	if err != nil {
+		return "", err
+	}
+	quota := strings.TrimSpace(string(data))
+	framework.Logf("pod %s/%s qos=%s path %q quota %q", pod.Namespace, pod.Name, pod.Status.QOSClass, cgPath, quota)
+	return quota, nil
+}
+
+func getContainerCFSQuota(pod *v1.Pod, ctnName string, isInit bool) (string, error) {
+	cgPath, err := makeCgroupPathForContainer(pod, ctnName, isInit)
+	if err != nil {
+		return "", err
+	}
+	data, err := os.ReadFile(filepath.Join(cgPath, "cpu.max"))
+	if err != nil {
+		return "", err
+	}
+	quota := strings.TrimSpace(string(data))
+	framework.Logf("pod %s/%s qos=%s cnt %s path %q quota %q", pod.Namespace, pod.Name, pod.Status.QOSClass, ctnName, cgPath, quota)
+	return quota, nil
+}
+
+const (
+	kubeCgroupRoot = "/sys/fs/cgroup/kubepods"
+)
+
+func pathCheckAndJoin(elem ...string) string {
+	ginkgo.GinkgoHelper()
+	gomega.Expect(elem).ToNot(gomega.BeEmpty(), "no path components given") // should never fail
+	// bootstrapping
+	retPath := elem[0]
+	rest := elem[1:]
+	for {
+		if _, err := os.Stat(retPath); os.IsNotExist(err) {
+			framework.Failf("path component %q does not exist: %v", retPath, err)
+		}
+		if len(rest) == 0 {
+			break
+		}
+		retPath = filepath.Join(retPath, rest[0])
+		rest = rest[1:]
+	}
+	return retPath
+}
+
+func makeCgroupPathForPod(pod *v1.Pod) string {
+	qosClass := ""
+	if pod.Status.QOSClass != v1.PodQOSGuaranteed {
+		qosClass = strings.ToLower(string(pod.Status.QOSClass))
+	}
+	return pathCheckAndJoin(kubeCgroupRoot, qosClass, "pod"+string(pod.UID))
+}
+
+func makeCgroupPathForContainer(pod *v1.Pod, ctnName string, isInit bool) (string, error) {
+	fullCntID, ok := findContainerIDByName(pod, ctnName, isInit)
+	if !ok {
+		return "", fmt.Errorf("cannot find status for container %q", ctnName)
```

Comment:
Reasoning to allow us to create targeted Kubernetes Aggregated Failures when troubleshooting flaky tests and distinguish failures between initContainers.

```suggestion
		if isInit {
		         return "", fmt.Errorf("cannot find status for init-container %q", ctnName)
		} else {
		         return "", fmt.Errorf("cannot find status for container %q", ctnName)
		}
```

Reply:
fixed slightly differently than suggested